packages/ui/lit-grid/src/dx-grid.ts

```diff
@@ -0,0 +1,252 @@
+        const colVisMinCell = this.getCell(colVisMin, j + rowVisMin);
+        if (colVisMinCell?.end) {
+          const { i: iStart } = posFromNumericNotation(colVisMinCell.pos);
+          acc.colExtMin = Math.min(acc.colExtMin, iStart);
+        }
+        const colVisMaxCell = this.getCell(colVisMax, j + rowVisMin);
+        if (colVisMaxCell?.end) {
+          const { i: iEnd } = posFromNumericNotation(colVisMaxCell.end);
+          acc.colExtMax = Math.max(acc.colExtMax, iEnd);
+        }
+        return acc;
+      },
+      { colExtMin: colVisMin, colExtMax: colVisMax },
+    );
+
+    const { rowExtMin, rowExtMax } = [...Array(colVisMax - colVisMin)].reduce(
+      (acc, _, i) => {
+        const rowVisMinCell = this.getCell(i + colVisMin, rowVisMin);
+        if (rowVisMinCell?.end) {
+          const { j: jStart } = posFromNumericNotation(rowVisMinCell.pos);
+          acc.rowExtMin = Math.min(acc.rowExtMin, jStart);
+        }
+        const rowVisMaxCell = this.getCell(i + colVisMin, rowVisMax);
+        if (rowVisMaxCell?.end) {
+          const { j: jEnd } = posFromNumericNotation(rowVisMaxCell.end);
+          acc.rowExtMax = Math.max(acc.rowExtMax, jEnd);
+        }
+        return acc;
+      },
+      { rowExtMin: rowVisMin, rowExtMax: rowVisMax },
+    );
+
+    return {
+      colVisMin,
+      colExtMin,
+      colVisMax,
+      colExtMax,
+      rowVisMin,
+      rowExtMin,
+      rowVisMax,
+      rowExtMax,
+    };
+  }
+
+  override render() {
+    const { colVisMin, colVisMax, rowVisMin, rowVisMax } = this.computeExtrema();
+
+    const visibleCols = colVisMax - colVisMin;
+    const visibleRows = rowVisMax - rowVisMin;
+
+    // TODO(thure): now compute the offset based on the extrema.
+    const offsetInline = colVisMin * colSize - this.posInline;
+    const offsetBlock = rowVisMin * rowSize - this.posBlock;
+
+    return html`<div role="none" class="dx-grid">
+      <div role="none" class="dx-grid__corner"></div>
+      <div role="none" class="dx-grid__columnheader">
+        <div
+          role="none"
+          class="dx-grid__columnheader__content"
+          style="transform:translate3d(${offsetInline}px,0,0);grid-template-columns:repeat(${visibleCols},${colSize}px);"
+        >
+          ${[...Array(visibleCols)].map((_, i) => {
+            return html`<div
+              role="gridcell"
+              style="inline-size:${colSize}px;block-size:${rowSize}px;grid-column:${i + 1}/${i + 2};"
+            >
+              ${colToA1Notation(colVisMin + i)}
+            </div>`;
+          })}
+        </div>
+      </div>
+      <div role="none" class="dx-grid__corner"></div>
+      <div role="none" class="dx-grid__rowheader">
+        <div role="none" class="dx-grid__rowheader__content" style="transform:translate3d(0,${offsetBlock}px,0);">
+          ${[...Array(visibleRows)].map((_, j) => {
+            return html`<div role="gridcell" style="block-size:${rowSize}px;grid-row:${j + 1}/${j + 2}">
```

Comment:
https://lit.dev/docs/templates/lists/#the-repeat-directive

Reply:
I encountered a noticeable performance issue using `repeat`: https://discord.com/channels/837138313172353095/1110334757126017076/1271594376749125662